fragment.go

```diff
@@ -1262,6 +1263,7 @@ func (f *Fragment) snapshot() error {
 	} else if err := bw.Flush(); err != nil {
 		return fmt.Errorf("flush: %s", err)
 	}
+	bw.Flush()
```

Comment:
this new line `bw.Flush()` seems extraneous. is there a reason to call it again?

somewhat related, i think the `else` statement before this makes it confusing to read. i.e. i think we could remove the `else` and just have a new `if` statement.